### DISCUSSION TITLE
Vertical Alignment

### DISCUSSION THREAD

**Discussion by quarterpound** [chakra-ui/chakra-ui#3822] [Q&A]
How would one align a Box vertically within a ```<Box height={"100vh"}><Box/>```

**Comment by chr-ge** [comment]
Take a look at this: https://blog.logrocket.com/13-ways-to-vertical-center/
There are a couple options.

**Using absolute position:**
```jsx
<Box position="relative">
  <Box
    height="40px"
    width="40px"
    position="absolute"
    top={0}
    right={0}
    left={0}
    bottom={0}
    margin="auto"
  ></Box>
</Box>
```

**50% translate -50%:**
```jsx
<Box position="relative">
  <Box
    height="40px"
    width="40px"
    position="absolute"
    top="50%"
    left="50%"
    transform="translateY(-50%, -50%)"
  ></Box>
</Box>
```

I would recommend using `Flex` for easiest vertical alignment.
